src/xpk/api/storage_crd.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: apiextensions.k8s.io/v1
```

Comment:
A general question: I'm new to k8s world, Do you think a configMap can be a better fit for this reason especially as we are not going to have a controller for the objects created based on this crd? What are pros and cons of using a CRD over ConfigMap?
@44past4

Reply:
If we decicded to go ahead with a CRD, do you think we can have a version for it so that if later we wanted to add new fields to it we can distinguish crd/objects versions there?

Let's say a CRD is created by a user with an older version of xpk, but upgrades xpk to a newer version that introduces new fields there. What will be the behavior?

Reply:
WDYT @44past4?

Reply:
The main pros of using CRDs over ConfigMaps are well defined structure, validation and documentation provided by CRD. Therefore in this case where we want to reuse the information about the storages in multiple places (listing storages, mounting them to workloads etc.) for me it makes sense to use CRD.

When it comes to versioning the definition below uses v1 version. As long as we will not be adding/removing required fields to CRD we should be able to keep v1. If we will need to add or remove some required fields we will need to bump the version to v2.

Reply:
Thanks Pawel!